keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,993 @@
+cluster.
+
+#### Cluster Manager
+
+An immutable field set by a cluster manager when this cluster resources
+is created by it. Each cluster manager instance should set a different
+values to this field.
+
+In addition, a predefined label with key "x-k8s.io/cluster-manager" needs
+to be added by the cluster manager upon creation. The value of the label
+MUST be the same as the name of the cluster manager. The purpose of this
+label is to make filter clusters from different cluster managers eaiser.
+
+### Status
+
+#### Version
+
+Kubernetes version of the cluster
+
+Versions of the kubernetes can let consumers understand the capability
+of the kubernetes, such as what API is supported.
+
+#### Resources
+
+Resources include capacity and allocatable, which is the sum of the
+capacity/allocatable of nodes on the cluster. It is useful for users to
+know the sizing of the cluster, and also schedulers to sort or bin pack
+the workload to clusters.
+
+#### Properties
+
+Name/value pairs to represent properties of the clusters. It could be a
+collection of ClusterProperty resources, but could also be info based on
+other implementations.
+
+#### Conditions
+
+Record cluster’s healthiness condition and easy to extend, conforming to
+metav1.Condition format.
+
+Predefined condition types:
+- Healthy conditions indicate the cluster is in a good state. (state:
+  True/False/Unknown). Healthiness can have different meanings in
+  different scenarios. We will have multiple condition types to define
+  healthiness:
+  - ControlPlaneHealthy is to define whether the controlplane of the
+    cluster is in the healthy state.
+    * apiserver/healthz
+    * controller-manager/healthz
+    * scheduler/healthz
+    * etcd/healthz
+  - AllNodesHealthy is to define if the nodes in the cluster are in a
+    healthy state. If one node is not healthy, the status of NodeHealthy
+    is set to false with the message indicating details. (todo
+    tolerance, it should be configurable)
+
+  It would be useful to collect the healthiness of other subsystems in
+  the cluster, e.g. network, dns, storage, or ingress. However, it is
+  not easy to collect that information in a common way with different
+  implementations of network or storage providers. We decide not to
+  include other subsystems healthiness conditions in the initial phase.
+- Joined: indicate the cluster is under management by the cluster manager.
+  The status of the cluster SHOULD be updated by the cluster manager under
+  this condition.
+
+## API Example
+
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ClusterInventory
+metadata:
+ name: generated-cluster-name
+ labels:
+   x-k8s.io/cluster-manager: some-cluster-manager
+spec:
+  displayName: some-cluster
+  clusterManager:
+    name: some-cluster-manager
+status:
+ version:
+   kubernetes: 1.28.0
+ resources: # total resources of a cluster
+   capacity:
+     cpu: 12
+     memory: 12Gi
+   allocatable:
+     cpu: 10
+     memory: 16Gi
```

Comment:
The way this resources are presented are of little help for a resource manager/scheduler.  12 cpu's could mean a node with 10 an another with 2. Then is impossible to make an informed scheduling decision based on the presented info

Reply:
yes, currently it is the sum of all nodes in a cluster to let consumer understand the overall capacity. Agree it is not quite helpful for scheduler. But resources has a map structure that it can also add info like `max_cpu_per_node` and `min_cpu_per_node`. My understanding is the scheduler on the multicluster level is to select a cluster, while the node selection is still done by the kube-scheduler, so the above info (`max_cpu_per_node`) would be enough for multicluster scheduler?

Reply:
well `max_cpu_per_node` can be confusing, if you have one "big" node and a bunch of small ones. 
That's why this conversation, I think is the most important on this review https://github.com/kubernetes/enhancements/pull/4323#discussion_r1377254904

Reply:
> well `max_cpu_per_node` can be confusing, if you have one "big" node and a bunch of small ones.

I think it could be reasonable to expect different "classes" of clusters to have different ClusterProperties to allow selective targeting if needed.  ARM versus AMD is another case of "nodes aren't the same", but if you're multi-cluster first, it's likely that clusters could be labelled in meaningful ways.

Reply:
lgtm